Assignment3/Q1.py

```diff
@@ -0,0 +1,128 @@
+"""The approach taken in this solution is based on a DFS-like search. The worst case running time
```

Comment:
Ooops! I was overthinking it. In case of NxN grid the running time is O(N^4) (N^2 calls of DFS which takes O(N^2) time N^2 vertices and at most 8N^2 edges, actually because each edge connectes two grid entried it might be even bounded by 4N^2 edges). In general case, I guess, it's O((MxN)^2) M-number of rows and N columns in a grid.